service/controller/v17/encrypter/kms/kms.go

```diff
@@ -51,24 +51,53 @@ func (e *Encrypter) EnsureCreatedEncryptionKey(ctx context.Context, customObject
 		return microerror.Mask(err)
 	}
 
+	var keyAlias string
+	{
+		clusterID := key.ClusterID(customObject)
+		keyAlias = toAlias(clusterID)
```

Comment:
Lets put this into the key package properly and use the function where we need it to safe these lines.

Reply:
I gave it a second thought and I don't think it's a good idea. This is an implementation detail. We shouldn't use this alias outside. That makes sense only for KMS implementation which is hidden behind `Encrypter` interface. Putting it in `key`  subpackage of `resource` package would be a leak of that implementation detail.

Reply:
Then lets have `key.go` in this package and have the interface the same at least so that it takes `customObject` and is not different to how we use key functions. Keeping it private makes sense though.